src/lm_datasets/datasets/de/de_laws.py

```diff
@@ -0,0 +1,334 @@
+import os
+import zipfile 
+import re
+import multiprocessing
+import requests
+import time
+import logging
+import copy
+import pprint
+import tiktoken
+
+import xml.etree.ElementTree as ET
+from bs4 import BeautifulSoup
+from tqdm import tqdm
+from typing import Dict, Union, Optional
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+from lm_datasets.datasets.base import BaseDataset, Availability, MILLION
+
+logger = logging.getLogger(__name__)
+
+
+def custom_request(url):
```

Comment:
Instead of retrying, I recommend adding a delay between the request.

Reply:
You mean, simply putting a timer between requests if the request failed?
Or a delay between every request?

Reply:
Rather add a delay between every request. The other approach will annoy the site you are scraping.